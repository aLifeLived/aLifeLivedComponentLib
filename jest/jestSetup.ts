/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
