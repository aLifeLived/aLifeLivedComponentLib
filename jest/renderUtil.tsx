import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
