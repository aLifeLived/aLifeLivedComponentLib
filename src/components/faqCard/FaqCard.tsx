import React, { useCallback, useState } from 'react';

// UI
import { Box } from '../../themes/box/Box';
import { SystemPressable } from '../../themes/systemPressable/SystemPressable';
import { Text } from '../../themes/text/Text';
import { Icon } from '../icon/Icon';

type FaqCardProps = {
  header: string;
  content: string;
};

export const FaqCard: React.FC<FaqCardProps> = ({ header, content }) => {
  const [isOpen, setIsOpen] = useState(false);
  const handleOnPress = useCallback(() => setIsOpen(!isOpen), [isOpen]);
  return (
    <SystemPressable
      borderColor={isOpen ? 'primary' : 'backgroundContrastLow'}
      borderWidth={1}
      borderRadius="medium"
      onPress={() => handleOnPress()}
    >
      <Box
        backgroundColor={isOpen ? 'white' : 'backgroundContrastLow'}
        borderTopLeftRadius="medium"
        borderTopRightRadius="medium"
        paddingVertical="m"
        paddingHorizontal="s"
        flexDirection="row"
        alignItems="center"
        justifyContent="space-between"
      >
        <Text variant="h4">{header}</Text>
        <Icon
          icon={isOpen ? 'chevron-down' : 'chevron-right'}
          color={isOpen ? 'primary' : 'backgroundContrastHigh'}
        />
      </Box>
      {isOpen && (
        <Box padding="s" marginTop="s">
          <Text variant="bodySmall">{content}</Text>
        </Box>
      )}
    </SystemPressable>
  );
};
