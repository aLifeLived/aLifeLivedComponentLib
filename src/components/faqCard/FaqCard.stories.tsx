import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { FaqCard } from './FaqCard';
import { Box } from '../../themes/box/Box';

const testText =
  'The information contained herein is preliminary, is provided for discussion purposes only, is only a summary of key information, is not complete, and does not contain certain material information about the Company.';

storiesOf('Cards', module)
  .addDecorator(Story => Story())
  .add('FAQ Card', () => (
    <Box margin="m">
      <Box margin="s">
        <FaqCard header="How do I delete a recording?" content={testText} />
      </Box>
      <Box margin="s">
        <FaqCard header="How do I record?" content={testText} />
      </Box>
      <Box margin="s">
        <FaqCard
          header="How should I be holding my device?"
          content={testText}
        />
      </Box>
    </Box>
  ));
