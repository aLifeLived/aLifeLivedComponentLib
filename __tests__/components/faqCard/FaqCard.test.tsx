import React from 'react';
import { fireEvent, render } from '../../../jest/renderUtil';

// Component
import { FaqCard } from '../../../src/components/faqCard/FaqCard';

const setup = () =>
  render(<FaqCard header="Test header" content="Test content" />);

describe('src >> components >> faqCard', () => {
  describe('Given container is NOT open', () => {
    test('Should match snapshot ', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find header text', () => {
      const { getByText } = setup();
      expect(getByText('Test header')).toBeDefined();
    });

    test('Should NOT find content text', () => {
      const { queryByText } = setup();
      expect(queryByText('Test content')).toBeNull();
    });
  });

  describe('Given container IS open', () => {
    test('Should match snapshot', () => {
      const { toJSON, getByText } = setup();
      fireEvent.press(getByText('Test header'));
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should find header text', () => {
      const { getByText } = setup();
      expect(getByText('Test header')).toBeDefined();
    });

    test('Should find content text', () => {
      const { getByText } = setup();
      fireEvent.press(getByText('Test header'));
      expect(getByText('Test content')).toBeDefined();
    });
  });
});
