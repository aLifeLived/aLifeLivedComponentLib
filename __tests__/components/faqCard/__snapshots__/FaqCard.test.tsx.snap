// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src >> components >> faqCard Given container IS open Should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderColor": "#67B6A8",
        "borderRadius": 8,
        "borderWidth": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontFamily": "SFUIText-Regular",
            "fontSize": 20,
            "lineHeight": 24,
          },
        ]
      }
    >
      Test header
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      fill="#67B6A8"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={512}
      vbWidth={448}
      width={16}
    >
      <RNSVGGroup
        fill={4284987048}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          fill={4284987048}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "marginTop": 8,
          "padding": 8,
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontFamily": "SFUIDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 20,
          },
        ]
      }
    >
      Test content
    </Text>
  </View>
</View>
`;

exports[`src >> components >> faqCard Given container is NOT open Should match snapshot  1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderColor": "#EFEFEF",
        "borderRadius": 8,
        "borderWidth": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFEFEF",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontFamily": "SFUIText-Regular",
            "fontSize": 20,
            "lineHeight": 24,
          },
        ]
      }
    >
      Test header
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      fill="#282828"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={512}
      vbWidth={320}
      width={16}
    >
      <RNSVGGroup
        fill={4280821800}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill={4280821800}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
